src/main/java/com/rybka/service/exchange/ExchangeService.java

```diff
@@ -18,19 +25,47 @@
 
     public CurrencyData loadCurrencyOf(String userBaseCurrency, String userTargetCurrency) {
         var exchangeResponse = connectorService.retrieveRates(userBaseCurrency);
-        var targetCurrency = retrieveTargetCurrency(exchangeResponse, userTargetCurrency);
+        var targetCurrency = retrieveTargetCurrency(exchangeResponse, userTargetCurrency).orElseThrow(() -> new MissedBaseCurrencyException("Target currency not found!"));
         return new CurrencyData(userBaseCurrency, targetCurrency.getKey(), targetCurrency.getValue());
     }
 
     public Double calculateTotal(Double rate, Double amount) {
         return rate * amount;
     }
 
-    private Map.Entry<String, Double> retrieveTargetCurrency(ExchangeResponse exchangeResponse, String userTargetCurrency) {
+    public ExchangeResultData retrieveTopCurrency(String base, String exchangeType, double count) {
+        var exchangeResponse = connectorService.retrieveRates(base);
+        var topCurrencyList = new ArrayList<TopCurrencyData>();
+
+        if (exchangeType.equals("SELL")) {
```

Comment:
Try to use map instead of if/else

Reply:
Map<String, BiFunction<Int, ExchangeResponse, List<TopCurrency>>

@PostConstruct <- read about this 
map.add('sell', sellHandler);
map.add('buy' buyHandler);

var topCurrencyList = mapExchangeTypeHandlers.get(exchangeType).accept(count, exchangeResponse);

Reply:
fixed